### PR TITLE
Custom Tags In Jekyll Using Ruby

### PR DESCRIPTION
**Pull request by yashwin**
Please don't merge this as there is some issue with this. It would be
great if you can help me with this.

### CODE REVIEW

File: _plugins/utils.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
+module Jekyll
+
```
Comment (subho007) [inline _plugins/utils.rb:13]: **[rubocop]**
- Extra empty line detected at module body beginning.

File: _plugins/utils.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        class AKIconTag < Liquid::Tag
```
Comment (subho007) [inline _plugins/utils.rb]: **[rubocop]**
- Use 2 (not 8) spaces for indentation. [:ref](https://github.com/bbatsov/ruby-style-guide#spaces-indentation)
- Missing top-level class documentation comment.

File: _plugins/utils.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
```
Comment (subho007) [inline _plugins/utils.rb]: **[rubocop]**
- Extra empty line detected at class body beginning.

File: _plugins/utils.rb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+              def initialize(tag_name,text,tokens)
```
Comment (subho007) [inline _plugins/utils.rb]: **[rubocop]**
- Use 2 (not 6) spaces for indentation. [:ref](https://github.com/bbatsov/ruby-style-guide#spaces-indentation)
- Space missing after comma. [:ref](https://github.com/bbatsov/ruby-style-guide#spaces-operators)

File: _plugins/utils.rb [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+                  super
```
Comment (subho007) [inline _plugins/utils.rb]: **[rubocop]**
- Use 2 (not 4) spaces for indentation. [:ref](https://github.com/bbatsov/ruby-style-guide#spaces-indentation)

File: _plugins/utils.rb [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+                  @text = text
+              end
+
+              def render(context)
```
Comment (subho007) [inline _plugins/utils.rb]: **[rubocop]**
- Method has too many lines. [14/10] [:ref](https://github.com/bbatsov/ruby-style-guide#short-methods)
- Unused method argument - `context`. If it's necessary, use `_` or `_context` as an argument name to indicate that it won't be used. You can also write as `render(*)` if you want the method to accept any arguments but don't care about them. [:ref](https://github.com/bbatsov/ruby-style-guide#underscore-unused-vars)

File: _plugins/utils.rb [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+                  case @text
```
Comment (subho007) [inline _plugins/utils.rb]: **[rubocop]**
- Use 2 (not 4) spaces for indentation. [:ref](https://github.com/bbatsov/ruby-style-guide#spaces-indentation)

File: _plugins/utils.rb [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+                  when /linkedin/
+                        "linkedin"
```
Comment (subho007) [inline _plugins/utils.rb]: **[rubocop]**
- Use 2 (not 6) spaces for indentation. [:ref](https://github.com/bbatsov/ruby-style-guide#spaces-indentation)
- Prefer single-quoted strings when you don't need string interpolation or special symbols. [:ref](https://github.com/bbatsov/ruby-style-guide#consistent-string-literals)

File: _plugins/utils.rb [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+                  when /twitter/
+                        "twitter"
```
Comment (subho007) [inline _plugins/utils.rb]: **[rubocop]**
- Use 2 (not 6) spaces for indentation. [:ref](https://github.com/bbatsov/ruby-style-guide#spaces-indentation)
- Prefer single-quoted strings when you don't need string interpolation or special symbols. [:ref](https://github.com/bbatsov/ruby-style-guide#consistent-string-literals)

File: _plugins/utils.rb [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+                  when /facebook/
+                        "facebook"
```
Comment (subho007) [inline _plugins/utils.rb]: **[rubocop]**
- Use 2 (not 6) spaces for indentation. [:ref](https://github.com/bbatsov/ruby-style-guide#spaces-indentation)
- Prefer single-quoted strings when you don't need string interpolation or special symbols. [:ref](https://github.com/bbatsov/ruby-style-guide#consistent-string-literals)

File: _plugins/utils.rb [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+                  when /github/
+                        "github"
```
Comment (subho007) [inline _plugins/utils.rb]: **[rubocop]**
- Use 2 (not 6) spaces for indentation. [:ref](https://github.com/bbatsov/ruby-style-guide#spaces-indentation)
- Prefer single-quoted strings when you don't need string interpolation or special symbols. [:ref](https://github.com/bbatsov/ruby-style-guide#consistent-string-literals)

File: _plugins/utils.rb [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+                  when /@/
+                        "envelope"
```
Comment (subho007) [inline _plugins/utils.rb]: **[rubocop]**
- Use 2 (not 6) spaces for indentation. [:ref](https://github.com/bbatsov/ruby-style-guide#spaces-indentation)
- Prefer single-quoted strings when you don't need string interpolation or special symbols. [:ref](https://github.com/bbatsov/ruby-style-guide#consistent-string-literals)

File: _plugins/utils.rb [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+                  else
+                        "globe"
```
Comment (subho007) [inline _plugins/utils.rb]: **[rubocop]**
- Use 2 (not 6) spaces for indentation. [:ref](https://github.com/bbatsov/ruby-style-guide#spaces-indentation)
- Prefer single-quoted strings when you don't need string interpolation or special symbols. [:ref](https://github.com/bbatsov/ruby-style-guide#consistent-string-literals)

File: _plugins/utils.rb [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+                  end
+              end
+        end
+
+        class AKLinkTag < Liquid::Tag
```
Comment (subho007) [inline _plugins/utils.rb]: **[rubocop]**
- Missing top-level class documentation comment.

File: _plugins/utils.rb [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+
```
Comment (subho007) [inline _plugins/utils.rb]: **[rubocop]**
- Extra empty line detected at class body beginning.

File: _plugins/utils.rb [hunk 16]
```diff
@@ -0,0 +1,46 @@
...
+              def initialize(tag_name,text,tokens)
```
Comment (subho007) [inline _plugins/utils.rb]: **[rubocop]**
- Use 2 (not 6) spaces for indentation. [:ref](https://github.com/bbatsov/ruby-style-guide#spaces-indentation)
- Space missing after comma. [:ref](https://github.com/bbatsov/ruby-style-guide#spaces-operators)

File: _plugins/utils.rb [hunk 17]
```diff
@@ -0,0 +1,46 @@
...
+                  super
```
Comment (subho007) [inline _plugins/utils.rb]: **[rubocop]**
- Use 2 (not 4) spaces for indentation. [:ref](https://github.com/bbatsov/ruby-style-guide#spaces-indentation)

File: _plugins/utils.rb [hunk 18]
```diff
@@ -0,0 +1,46 @@
...
+                  @text = text
+              end
+
+              def render(context)
```
Comment (subho007) [inline _plugins/utils.rb]: **[rubocop]**
- Unused method argument - `context`. If it's necessary, use `_` or `_context` as an argument name to indicate that it won't be used. You can also write as `render(*)` if you want the method to accept any arguments but don't care about them. [:ref](https://github.com/bbatsov/ruby-style-guide#underscore-unused-vars)

File: _plugins/utils.rb [hunk 19]
```diff
@@ -0,0 +1,46 @@
...
+                  if @text.include? "@"
```
Comment (subho007) [inline _plugins/utils.rb]: **[rubocop]**
- Use 2 (not 4) spaces for indentation. [:ref](https://github.com/bbatsov/ruby-style-guide#spaces-indentation)
- Prefer single-quoted strings when you don't need string interpolation or special symbols. [:ref](https://github.com/bbatsov/ruby-style-guide#consistent-string-literals)

File: _plugins/utils.rb [hunk 20]
```diff
@@ -0,0 +1,46 @@
...
+                        "mailto://#{@text}"
```
Comment (subho007) [inline _plugins/utils.rb]: **[rubocop]**
- Use 2 (not 6) spaces for indentation. [:ref](https://github.com/bbatsov/ruby-style-guide#spaces-indentation)

File: _plugins/utils.rb [hunk 21]
```diff
@@ -0,0 +1,46 @@
...
+                  else
+                      "#{@text}"
```
Comment (subho007) [inline _plugins/utils.rb]: **[rubocop]**
- Use 2 (not 4) spaces for indentation. [:ref](https://github.com/bbatsov/ruby-style-guide#spaces-indentation)
- Prefer `to_s` over string interpolation.

File: _plugins/utils.rb [hunk 22]
```diff
@@ -0,0 +1,46 @@
...
+                  end
+              end
+        end
+end
+
+Liquid::Template.register_tag('ak_icon',Jekyll::AKIconTag)
```
Comment (subho007) [inline _plugins/utils.rb]: **[rubocop]**
- Space missing after comma. [:ref](https://github.com/bbatsov/ruby-style-guide#spaces-operators)

File: _plugins/utils.rb [hunk 23]
```diff
@@ -0,0 +1,46 @@
...
+Liquid::Template.register_tag('ak_link',Jekyll::AKLinkTag)
```
Comment (subho007) [inline _plugins/utils.rb]: **[rubocop]**
- Space missing after comma. [:ref](https://github.com/bbatsov/ruby-style-guide#spaces-operators)

**Comment by subho007** [comment]
Click here if you want to create rubocop's auto-correct Pull Request 
[](https://sideci.com/github_repositories/55060189/auto_correct/rubocop/pull_request/21/branch/custom-tags)

See the documentation if you want to get more information about auto-correct.
[Document](https://sideci.com/ja/docs/autofix-rubocop)

**Comment by dhilipsiva** [comment]
LOL for auto comments! 😆

File: _plugins/utils.rb [hunk 24]
```diff
@@ -0,0 +1,43 @@
+module Jekyll
+  class AKIconTag < Liquid::Tag
```
Comment (subho007) [inline _plugins/utils.rb]: **[rubocop]**
- Missing top-level class documentation comment.

File: _plugins/utils.rb [hunk 25]
```diff
@@ -0,0 +1,43 @@
...
+
```
Comment (subho007) [inline _plugins/utils.rb]: **[rubocop]**
- Extra empty line detected at class body beginning.

File: _plugins/utils.rb [hunk 26]
```diff
@@ -0,0 +1,43 @@
...
+    def initialize(tag_name, text, tokens)
+      super
+      @text = text
+    end
+
+    def render(context)
```
Comment (subho007) [inline _plugins/utils.rb]: **[rubocop]**
- Method has too many lines. [12/10] [:ref](https://github.com/bbatsov/ruby-style-guide#short-methods)
- Unused method argument - `context`. If it's necessary, use `_` or `_context` as an argument name to indicate that it won't be used. You can also write as `render(*)` if you want the method to accept any arguments but don't care about them. [:ref](https://github.com/bbatsov/ruby-style-guide#underscore-unused-vars)

File: _plugins/utils.rb [hunk 27]
```diff
@@ -0,0 +1,43 @@
...
+      when /twitter/
+        'twitter'
+      when /@/
+        'envelope'
+      else
+        'globe'
+      end
+    end
+  end
+
```
Comment (subho007) [inline _plugins/utils.rb]: **[rubocop]**
- Trailing whitespace detected. [:ref](https://github.com/bbatsov/ruby-style-guide#no-trailing-whitespace)

File: _plugins/utils.rb [hunk 28]
```diff
@@ -0,0 +1,43 @@
...
+        'twitter'
+      when /@/
+        'envelope'
+      else
+        'globe'
+      end
+    end
+  end
+  
+  class AKLinkTag < Liquid::Tag
```
Comment (subho007) [inline _plugins/utils.rb]: **[rubocop]**
- Missing top-level class documentation comment.

File: _plugins/utils.rb [hunk 29]
```diff
@@ -0,0 +1,43 @@
...
+
```
Comment (subho007) [inline _plugins/utils.rb]: **[rubocop]**
- Extra empty line detected at class body beginning.

File: _plugins/utils.rb [hunk 30]
```diff
@@ -0,0 +1,43 @@
...
+    def initialize(tag_name, text, tokens)
+      super
+      @text = text
+    end
+
+    def render(context)
```
Comment (subho007) [inline _plugins/utils.rb]: **[rubocop]**
- Unused method argument - `context`. If it's necessary, use `_` or `_context` as an argument name to indicate that it won't be used. You can also write as `render(*)` if you want the method to accept any arguments but don't care about them. [:ref](https://github.com/bbatsov/ruby-style-guide#underscore-unused-vars)

File: _plugins/utils.rb [hunk 31]
```diff
@@ -0,0 +1,43 @@
...
+      if @text.include? "@"
```
Comment (subho007) [inline _plugins/utils.rb]: **[rubocop]**
- Prefer single-quoted strings when you don't need string interpolation or special symbols. [:ref](https://github.com/bbatsov/ruby-style-guide#consistent-string-literals)

File: _plugins/utils.rb [hunk 32]
```diff
@@ -0,0 +1,43 @@
...
+        "mailto://#{@text}"
+      else
+          "#{@text}"
```
Comment (subho007) [inline _plugins/utils.rb]: **[rubocop]**
- Use 2 (not 4) spaces for indentation. [:ref](https://github.com/bbatsov/ruby-style-guide#spaces-indentation)
- Prefer `to_s` over string interpolation.

File: _plugins/utils.rb [hunk 33]
```diff
@@ -0,0 +1,29 @@
+module Jekyll
+  module AKUtils
```
Comment (subho007) [inline _plugins/utils.rb:2]: **[rubocop]**
- Missing top-level module documentation comment.

File: _plugins/utils.rb [hunk 34]
```diff
@@ -0,0 +1,29 @@
...
+
```
Comment (subho007) [inline _plugins/utils.rb]: **[rubocop]**
- Extra empty line detected at module body beginning.

File: _plugins/utils.rb [hunk 35]
```diff
@@ -0,0 +1,29 @@
...
+    def ak_icon(text)
```
Comment (subho007) [inline _plugins/utils.rb:3]: **[rubocop]**
- Method has too many lines. [12/10] [:ref](https://github.com/bbatsov/ruby-style-guide#short-methods)

File: _plugins/utils.rb [hunk 36]
```diff
@@ -0,0 +1,29 @@
...
+        'twitter'
+      when /@/
+        'envelope'
+      else
+        'globe'
+      end
+    end
+
+    def ak_link(text)
+      if text.include? "@"
```
Comment (subho007) [inline _plugins/utils.rb]: **[rubocop]**
- Prefer single-quoted strings when you don't need string interpolation or special symbols. [:ref](https://github.com/bbatsov/ruby-style-guide#consistent-string-literals)

File: _plugins/utils.rb [hunk 37]
```diff
@@ -0,0 +1,29 @@
...
+        "mailto://#{text}"
+      else
+          "#{text}"
```
Comment (subho007) [inline _plugins/utils.rb]: **[rubocop]**
- Use 2 (not 4) spaces for indentation. [:ref](https://github.com/bbatsov/ruby-style-guide#spaces-indentation)
- Prefer `to_s` over string interpolation.

**Comment by subho007** [comment]
Created Auto-correct Pull Request from [SideCI](https://sideci.com/)

[https://github.com/appknox/appknox.github.io/pull/22](https://github.com/appknox/appknox.github.io/pull/22)

File: _plugins/utils.rb [hunk 38]
```diff
@@ -0,0 +1,28 @@
+module Jekyll
+  module AKUtils
+    def ak_icon(text)
```
Comment (subho007) [inline _plugins/utils.rb:3]: **[rubocop]**
- Method has too many lines. [12/10] [:ref](https://github.com/bbatsov/ruby-style-guide#short-methods)

File: _plugins/utils.rb [hunk 39]
```diff
@@ -0,0 +1,27 @@
+=begin
```
Comment (subho007) [inline _plugins/utils.rb]: **[rubocop]**
- Do not use block comments. [:ref](https://github.com/bbatsov/ruby-style-guide#no-block-comments)

File: _plugins/utils.rb [hunk 40]
```diff
@@ -0,0 +1,27 @@
...
+Utils
+=end
+module Jekyll
+  module AKUtils
```
Comment (subho007) [inline _plugins/utils.rb:2]: **[rubocop]**
- Missing top-level module documentation comment.

**Pull request merged**